src/frontend/store/data-context.js

```diff
@@ -0,0 +1,44 @@
+
+/**
+ * WordPress dependencies
+ */
+import { createContext, useContext, useState, useEffect } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { format } from '@wordpress/date';
+
+const EventDataContext = createContext( [] );
+
+export function useEventData() {
+	const context = useContext( EventDataContext );
+	if ( context === undefined ) {
+		throw new Error( 'useEvents must be used within a Provider' );
+	}
+	return context;
+}
+
+export function EventDataProvider( { children, data } ) {
+	const [ eventData, setEventData ] = useState( [] );
+	// we bail out early if data is hydrated from the server.
+	if ( data ) {
+		return (
+			<EventDataContext.Provider value={ data }>
+				{ children }
+			</EventDataContext.Provider>
+		);
+	}
+	useEffect(() => {
+		const currentDate = new Date();
+		const fromDate = new Date( currentDate.getFullYear(), currentDate.getMonth(), 1 );
+
+		apiFetch( {
+			path: `/wp/v2/meetings/from/${ format( 'Y-m-d', fromDate ) }?per_page=12`,
```

Comment:
the value 12 is what I brought from the logic within the PHP code.

Reply:
also depending on the size of data 12 per page could mean several months in advance or simply this week, so the API might be changed to some sort of from-to, and we can fetch more months as long the user is changing the page.